### PR TITLE
fix(executor): validate session Unix identity before claim

### LINKED ISSUE #2244: Decouple validateSessionUnixUsername from branch RBAC (stale Unix identity executes in open-access mode)
## Summary

`validateSessionUnixUsername()` is nested inside `branchRbacEnabled` conditionals at two call
sites, but the property it protects is independent of branch RBAC. With `branch_rbac: false`
and `unix_user_mode: delegated|strict`, a session whose creator's `unix_username` has since
changed still executes as the **stale** Unix user. With RBAC enabled, the same request is
refused.

## Where

- `apps/agor-daemon/src/register-hooks.ts:1011` — `messages.create`
- `apps/agor-daemon/src/register-hooks.ts:2925` — `tasks.create`

The validator's own contract is mode-independent (`apps/agor-daemon/src/utils/branch-authorization.ts:1206-1246`):
it compares `session.unix_username` against the creator's *current* value and refuses on drift,
because "SDK session data is stored in the original user's home directory and cannot be
accessed." Nothing about branch permissions.

The correct pattern already exists in the same file — session creation widens the condition at
`register-hooks.ts:2640`:

```ts
...(branchRbacEnabled || executionMode.requiresUserUnixUsername
  ? [setSessionUnixUsername(usersRepository, executionMode.unixUserMode)]
  : []),
```

## Why this isn't a one-line hoist

Unlike `ensureSessionImmutability` (hoisted in #2239), this validator hard-depends on
`loadSession` having run — `branch-authorization.ts:1222-1224` throws
`'loadSession hook must run before validateSessionUnixUsername'`. A correct fix must make
`resolveSessionContext()` + `loadSession()` unconditional on `messages.create` and
`tasks.create`, which adds a DB read to **every prompt** in the default configuration.

That is a behaviour *and* performance change on the hottest path in the product, which is why
it was deliberately kept out of #2239 rather than absorbed into it.

## Severity — deliberately not overstated

- **Not** triggerable at will by an unprivileged user: requires an admin changing a user's
  `unix_username` after session creation.
- A failure to fail-closed on identity drift, **not** a privilege escalation.
- **Pre-existing on `main`** — neither introduced nor worsened by #2239.

## Supporting detail

- `/tasks/:id/run` (`apps/agor-daemon/src/register-routes.ts:1740`) is itself `branchRbacEnabled &&`
  gated and performs only a branch-permission check — no identity-consistency check, so it does
  not cover the gap.
- `apps/agor-daemon/src/register-services.ts:936` feeds `session.unix_username` straight into
  `resolveUnixUserForImpersonation`.

## Provenance

Found by a `gpt-5.6-sol` audit of all 38 `branchRbacEnabled` conditionals in `register-hooks.ts`,
requested while reviewing #2239. Those two were the only additional mode-independent guards
found incorrectly nested; the rest are genuine branch visibility/permission enforcement that
correctly disappears in open-access mode.

### PR DESCRIPTION
**Pull request by richardfogaca**
## Linked issue

Fixes #2244

## Summary

- validate the session creator's current Unix identity at the common task-launch boundary
- apply the guard in `delegated` and `strict` modes independently of branch RBAC
- refuse stale, missing, or tenant-hidden creators before task claim or executor spawn

## Why / context

Session Unix identity validation was coupled to branch authorization. With branch RBAC disabled, a session could retain a stale Unix username and still launch work as the wrong operating-system user.

Moving the check to the shared launch fence keeps access-control policy and process identity enforcement separate. Refused tasks remain pending and unclaimed.

## Implementation notes

- Reuses a shared session/creator identity assertion from both the service hook and task-launch path.
- Performs the tenant-scoped creator lookup after confirming the task is pending and before the atomic dispatch claim.
- Preserves existing behavior for `simple` and `insulated` Unix modes without an additional user lookup.
- Covers prompt, explicit run, reactive and fleet queue drains, callbacks, and schedules through their common launch path.

## Validation / test plan

- [x] Focused Unix identity and launch suites: 111 passed
- [x] Target PostgreSQL/RLS matrix: 8 passed, 0 failed, 0 skipped
- [x] Repository PostgreSQL integration: 124 passed, 0 failed, 0 skipped
- [x] Daemon regression suite: 2,439 passed
- [x] Daemon typecheck
- [x] Biome on changed files and pre-commit hook
- [x] `pnpm check:multitenancy-boundaries`
- [x] `git diff --check`

## Risks / rollout / rollback

Identity validation is intentionally point-in-time; a username can still change after validation. A refused queued task remains pending until the identity is restored or a new session is used.

No schema, migration, dependency, or configuration changes are required. Rollback is a commit revert.

## Out of scope / follow-ups

This change does not add identity generations, row locking, or an external executor process integration test.

### CODE REVIEW

**Comment by mistercrunch** [comment]
I traced the affected launch path end-to-end against current `origin/main` (`5936df9a`) and the exact PR head (`ce45848e`). The central design is directionally right: Unix identity drift is a launch invariant, not a branch-RBAC or transport invariant, and `spawnTaskExecutor -> launchPendingTask -> tenant-scoped creator lookup -> atomic Session+Task dispatch claim` is the correct common boundary. REST/Socket prompt, MCP, gateway, scheduler, callbacks, reactive drains, and the fleet queue worker all converge there; the repository remains the correct owner of queue order, Session-first locking, expected-state checks, and Task/Session projection.

I found one blocker in this PR and one important pre-existing adjacent boundary issue:

### Blocker: required-mode `null === null` still crosses the claim fence

Copilot's inline comment is correct. `assertSessionUnixIdentityMatchesCreator()` normalizes both absent values to `null` and checks only equality. In `delegated`/`strict`, a legacy/imported/session-created-before-mode-switch record with both Session and creator usernames absent therefore passes, wins the dispatch claim, and only fails later in `resolveUnixUserForImpersonation()`. The Task becomes failed instead of remaining pending/unclaimed, contrary to the purpose of this change.

Please assert that the stamped Session username is present whenever `unixUserModeRequiresUsername(mode)` is true, before the claim, and add `created` + `queued` delegated/strict tests for present creator + null/null identity.

### Important pre-existing issue: prompt authorization is after the side effect

The common launch stack is not yet fully correctly ordered for external `/sessions/:id/prompt` calls. The route authenticates and loads the Session (view-tier under RBAC), then writes through `TaskRepository.createPending()` and can win `claimDispatchAndProjectSession()`. It only reaches `ensureCanPromptInSession` afterward through the `sessions.patch` in `onClaimed`. A view-only caller can therefore persist a Task and move Task -> `dispatching` / Session -> `running`, then receive `Forbidden` before executor spawn.

Relevant locations at PR head:

- admission/claim: `apps/agor-daemon/src/register-routes.ts:1427-1630`, especially `1596-1603` and `1216-1244`
- late authorization via Session patch: `apps/agor-daemon/src/register-routes.ts:1272-1313`
- intended prompt gate: `apps/agor-daemon/src/register-hooks.ts:2914-2927`

This is pre-existing on main, not introduced by #2255, so I would accept it as an immediate follow-up rather than expanding this PR, but it should be tracked/fixed urgently. External action authorization belongs at admission before Task persistence/claim; the launch-time Unix identity fence should remain at the shared pre-claim boundary.

### Non-blocking hardening

Raw REST/Socket prompt ingress has only TypeScript annotations plus a truthiness check, while MCP has a proper Zod DTO schema. A follow-up should add shared runtime validation (including prompt type/size and option enums) at the REST/Socket boundary. That should not replace the domain identity check or repository invariants.

I also simulated merging onto current main: no textual conflicts, and #2239's unconditional Session identity immutability/update coverage is preserved. Focused tests on the integrated tree passed (112 daemon tests), the multitenancy boundary check passed, and `git diff --check` was clean.

**Verdict:** request changes for the delegated/strict null-identity case. Otherwise the core layering introduced here is pragmatic and correct; no broad refactor is needed.

File: apps/agor-daemon/src/utils/session-unix-identity.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+}: PendingTaskLaunchInput): Promise<Task> {
+  if (!isTaskPendingDispatch(task)) return task;
+
+  const dispatchClaim = await runWithTenantDatabaseScope(db, tenantId, async (tenantDb) => {
+    await assertTenantWritable(tenantDb, tenantId);
+    const unixUserMode = execution?.unix_user_mode ?? 'simple';
+    if (unixUserModeRequiresUsername(unixUserMode)) {
+      // Fresh point-in-time fence only: user updates are not locked across the later spawn.
+      await assertSessionUnixIdentityMatchesCreator(session, new UsersRepository(tenantDb));
+    }
```
[the remark this answers is not in the document]
Comment (richardfogaca) [reply apps/agor-daemon/src/utils/session-unix-identity.ts:80]: Hey, Richard's agent here. Fixed this in 783cb6e2 by enforcing the existing required-username policy at the shared pre-claim fence. The delegated/strict matrix now covers both created and queued null/null identities across RBAC settings and verifies they remain pending and unclaimed.

**Comment by richardfogaca** [comment]
Richard's agent here: thanks for the review. 

I fixed the required-mode null/null blocker at the shared launch fence in 783cb6e2 and added the requested created/queued coverage. I kept the two adjacent findings out of this PR: prompt authorization ordering is a pre-existing admission issue, while shared REST/Socket runtime validation is broader ingress hardening; both need separate changes rather than widening this Unix-identity fix.
